### PR TITLE
[Windows] Stabilizing QuarkusPropertiesDistTest

### LINKED ISSUE #35496: `QuarkusPropertiesDistTest` fails on Windows
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

testsuite

### Describe the bug

The `QuarkusPropertiesDistTest` is testing some areas that can easily break on Windows, there's even a [specific test for Windows](https://github.com/keycloak/keycloak/blob/4a13875fb25cfb0cc85aab546daedd273ca48a32/quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/QuarkusPropertiesDistTest.java#L176-L185). However, the tests [seem to fail on Windows](https://github.com/vmuzikar/keycloak/actions/runs/12117249404/job/33779714328). We don't even run it in GHA.

We should:
* Fix the test on Windows.
* Add the `@Tag(DistributionTest.WIN)` tag to the test class so it's ran in GHA.

### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

`QuarkusPropertiesDistTest` works on Windows

### Actual behavior

`QuarkusPropertiesDistTest` doesn't work on Windows

### How to Reproduce?

Run `QuarkusPropertiesDistTest` on Windows

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Pepo48**
Closes: #35496

### CODE REVIEW

File: quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/QuarkusPropertiesDistTest.java [hunk 1]
```diff
@@ -68,19 +70,38 @@ void testPropertyEnabledAtRuntime(CLIResult cliResult) {
...
+    @DisabledOnOs(value = { OS.WINDOWS })
     @Order(4)
     void testIgnoreQuarkusSystemPropertyAtBuild(CLIResult cliResult) {
         cliResult.assertMessage("Keycloak is the best");
         cliResult.assertBuild();
     }
 
+    @Test
+    @Launch({"-Dquarkus.log.handler.console.\\\"console-2\\\".enable=false", "build"})
+    @DisabledOnOs(value = { OS.LINUX, OS.MAC }, disabledReason = "Different handling of quotes within arguments on Windows")
```
Comment (shawkins) [inline quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/QuarkusPropertiesDistTest.java:98]: Could this, and the other similar ones, just be EnabledOnOs instead?

Comment (Pepo48) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/QuarkusPropertiesDistTest.java:98]: Yes, both ways work. Intuitively, I also used EnabledOnOs at first, but then I realized that historically we've tried to use just one approach per test class. I guess it's a convention with the benefit of saving one unecessary import. But maybe I'm making it up 😄.

Anyway, I can certainly adjust it if necessary. 😉.

**Review by vmuzikar — COMMENTED** [review]
@Pepo48 The changes look good but CI seems to fail. Could you please check it?

**Comment by Pepo48** [comment]
@vmuzikar https://github.com/keycloak/keycloak/compare/30e0402402050d829a1b645d60c3183d4c482378..9638eb261809dd0bdefa61103b2062a35d8e3b91 is going to fix the failure. Disk letter on GHA runners is different, therefore I didn't spot it locally at first. By being more explicit, we can prevent ambiguity here.

**Review by vmuzikar — DISMISSED** [review]
Looks good, thanks.

File: quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/QuarkusPropertiesDistTest.java (hunk 1 above)
Comment (shawkins) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/QuarkusPropertiesDistTest.java:98]: EnabledOn would be more future proof as several *nix flavors are separated out.

Comment (Pepo48) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/QuarkusPropertiesDistTest.java]: Fair enough - addressed in 9f541810e43885f1b703152ed67f74ea86ef6e95.

If it makes sense, I can prepare a follow up in order to make this approach consistent all over the testsuite.

Thanks!

Comment (shawkins) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/QuarkusPropertiesDistTest.java]: If you see that in more places it seems good to simplify.

**Review by shawkins — APPROVED** [review]
LGTM, thanks @Pepo48

**Review by vmuzikar — APPROVED** [review]
Thanks, LGTM.

**Pull request merged**